// Proto file for movie info service. Note this is gRPC proto syntax (not Go)
syntax = "proto3";

option go_package = "github.com/cbeam3902/CloudNativeCourse/lab5/movieapi";

package movieapi;

service MovieInfo {
	// Sends a requeest for movie info
	rpc GetMovieInfo (MovieRequest) returns (MovieReply) {}
	rpc SetMovieInfo (MovieData) returns (Status) {}
}

message MovieData {
	string title = 1;
	int32 year = 2;
	string director = 3;
	repeated string cast = 4;
}

message Status {
	int32 code = 1;
}

// The request message containing movie name
message MovieRequest {
	string title = 1;
}

// The response message containining movie info
message MovieReply {
	int32 year = 1;
	string director = 2;
	repeated string cast = 3;
	
}

